block-report.js

```diff
@@ -14,118 +14,127 @@ import xlsx from 'xlsx';
-            continue;
+            return;
         }
 
+        const entryReport = {
+            blocks: [],
+            links: [],
+        };
         const mdast = await getMdast(markdown);
         const tableMap = getTableMap(mdast);
 
         for (let j = 0; j < tableMap.length; j++) {
             const table = tableMap[j];
             const { blockName, options } = table;
+            const blockReport = {
+                name: blockName,
+                index: j,
+                links: [],
+            };
             if (options) {
                 const variant = blockName + ' (' + options.join(', ') + ')';
-                variants[entry] = variants[entry] || {};
-                variants[entry][variant] = (variants[entry][variant] || 0) + 1;
+                blockReport.variant = variant;
                 totals.variants[variant] = (totals.variants[variant] || 0) + 1;
             }
-            blocks[entry][blockName] = (blocks[entry][blockName] || 0) + 1;
             totals.blocks[blockName] = (totals.blocks[blockName] || 0) + 1;
 
             const links = getNodesByType(table.table, 'link');
-            const name = `${entry},${blockName},${j}`;
-            blockLinks[name] = blockLinks[name] || [];
-            links.forEach((link) => {
-                const { url } = link;
-                blockLinks[name].push(url);
+            links.forEach(({ url }) => {
+                blockReport.links.push(url);
             });
+            entryReport.blocks.push(blockReport);
         };
 
         const links = getNodesByType(mdast, 'link');
-        if (links) {
-            links.forEach((link) => {
-                const { url } = link;
-                allLinks[entry] = allLinks[entry] || [];
-                allLinks[entry].push(url);
-            });
-        }
-
-        console.log(`${i}/${entries.length}`, entry, blocks[entry]);
-    }
-
-    return { blocks, variants, allLinks, blockLinks, totals };
+        links?.forEach((link) => {
+            const { url } = link;
+            entryReport.links.push(url);
+        });
+        report[entry] = entryReport;
+        console.log(`${i}/${entries.length}`, entry, entryReport.blocks.map(block => block.name));
+    });
+
+    return { report, totals };
 }
 
-export async function createReports(project, site, report) {
-    const { blocks, variants, allLinks, blockLinks, totals } = report;
+export async function createReports(project, site, { report, totals }) {
     const allBlocks = Object.keys(totals.blocks).sort();
     const ws_data = [['Path', 'URL', ...allBlocks]];
 
-    for (const entry in blocks) {
-        const blockValues = allBlocks.map((key) => blocks[entry][key] || 0);
-        const row = [entry, `${site}${entry}`, ...blockValues];
+    // Block Entries
+    console.log('creating block entries sheet');
+    for (const entry in report) {
+        const blocks = allBlocks.map((key) => report[entry].blocks.reduce((acc, block) => {
```

Comment:
block level function calls here might add some readability, if you thought it prudent. Would add a few lines with `reutrn`s though.

Reply:
Maybe after we run some more reports we can refactor this.